utils_cv/tracking/model.py

```diff
@@ -90,10 +95,107 @@ def write_video(results: Dict[int, List[TrackingBbox]],
     print(f"Output saved to {output_video}.")
 
 
-def savetxt_results(results: Dict[int, List[TrackingBbox]],
-                    exp_name: str = 'results',
-                    root_path: str = None,
-                    result_filename: str = 'results.txt') -> str:
+def plot_single_frame(
+    results: Dict[int, List[TrackingBbox]], input_video: str, frame_id: int
+) -> None:
+    """ 
+    Plot the bounding box and id on a wanted frame. Display as image to front end. 
+
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        input_video: path to the input video
+        frame_id: frame_id for frame to show tracking result
+    """
+    results = OrderedDict(sorted(results.items()))
+
+    # read video and initialize new tracking video
+    video = cv2.VideoCapture()
+    video.open(input_video)
+
+    image_width = int(video.get(cv2.CAP_PROP_FRAME_WIDTH))
+    image_height = int(video.get(cv2.CAP_PROP_FRAME_HEIGHT))
+    fourcc = cv2.VideoWriter_fourcc(*"MP4V")
+    frame_rate = int(video.get(cv2.CAP_PROP_FPS))
+
+    # assign bbox color per id
+    unique_ids = list(
+        set([bb.track_id for frame in results.values() for bb in frame])
+    )
+    color_map = assign_colors(unique_ids)
+
+    # Extract wanted frame:
+    video.set(cv2.CAP_PROP_POS_FRAMES, frame_id)
+    _, cur_image = video.read()
+
+    # Extract tracking results for wanted frame, and draw bboxes
+    cur_tracks = results[frame_id]
+    if len(cur_tracks) > 0:
+        cur_image = draw_boxes(cur_image, cur_tracks, color_map)
+
+    im = Image.fromarray(cur_image, "RGB")
+    b, g, r = im.split()
+    im = Image.merge("RGB", (r, g, b))
+    im.save("test_" + str(frame_id) + ".png")
+
+    IPython.display.display(im)
+
+
+def display_video(
+    results: Dict[int, List[TrackingBbox]], input_video: str
+) -> None:
+    """ 
+     Plot the predicted tracks on the input video. Displays to front-end as sequence of images stringed together in a video.
+
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        input_video: path to the input video        
+    """
+
+    results = OrderedDict(sorted(results.items()))
+
+    # assign bbox color per id
+    unique_ids = list(
+        set([bb.track_id for frame in results.values() for bb in frame])
+    )
+    color_map = assign_colors(unique_ids)
+
+    # read video and initialize new tracking video
+    video_reader = decord.VideoReader(input_video)
+
+    # set up ipython jupyter display
+    d_video = IPython.display.display("", display_id=1)
+
+    # Read each frame
+    frame_idx = 0
+    while True:
+        try:
```

Comment:
- Is it necessary to use "try..except" here? It's bad practice and hides unintended errors/warnings.
- [minor] if you can try to avoid the "while" loop, but maybe get the number of frames from len(results) if possible and then do for frame_idx in range(len(results))